### PR TITLE
[protocol] Implement Get Host State

### LINKED ISSUE #44: Implement "Get Host State" message
This is a straight-forward device state request defined in section 6.21 of the Challenge Protocol Specification.

### PR DESCRIPTION
**Pull request by kesyog**
Add serializer, deserializer, and fuzz tests for Get Host State request
and response messages (6.22 in the Cerberus Protocol). This message is
used to query the host processor reset state.

Fixes #44

Signed-off-by: Kesavan Yogeswaran <hikes@google.com>

### CODE REVIEW

File: src/protocol/get_host_state.rs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    fn to_wire<W: Write>(&self, mut w: W) -> Result<(), ToWireError> {
+        w.write_le(self.port_id)?;
+        Ok(())
+    }
+}
+
+make_fuzz_safe! {
+    /// The [`GetHostState`] response.
+    #[derive(Clone, Copy, PartialEq, Eq, Debug)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
```
Comment (mcy) [inline src/protocol/get_host_state.rs:70]: Uh oh, looks like `rustfmt` got confused. I bet you copy-pasted this from a place where I got this wrong, hehe.

File: src/protocol/mod.rs [hunk 2]
```diff
@@ -203,6 +210,10 @@ impl From<u8> for CommandType {
             0x02 => CommandType::DeviceCapabilities,
             0x03 => CommandType::DeviceId,
             0x04 => CommandType::DeviceInfo,
+            0x81 => CommandType::GetDigests,
+            0x82 => CommandType::GetCert,
+            0x83 => CommandType::Challenge,
```
Comment (mcy) [inline src/protocol/mod.rs:215]: Whoops.

Mind putting this in its own commit to the effect of "[protocol] Add missed entries to `ComandType::from()`"?

`git add -p` is your friend here. =)

**Review by mcy — APPROVED** [review]
Two nits, but otherwise this is really awesome. I'm guessing this was mostly copy-and-pasted, which is fine, but I love to see that it's straightforward to implement protocol messages!

**Comment by mcy** [comment]
Ah. It appears a recent refactor I did broke you, sorry. =P

Should be a matter of `s/(To/From)WireError/wire::Error/g`, plus or minus appropriate `use` statements.

File: src/protocol/get_host_state.rs (hunk 1 above)
Comment (kesyog) [reply src/protocol/get_host_state.rs:66]: Fixed. I had thought some macro parsing oddity earlier, but turns out I was misreading the compiler error I was fixing at the time.

File: src/protocol/mod.rs (hunk 2 above)
Comment (kesyog) [reply src/protocol/mod.rs:213]: Done

**Comment by kesyog** [comment]
Definitely was largely a copy-paste-rename job, but it helped to get a (little) better feel for the codebase.

I've rebased to include the latest wire error changes and fixed the feedback. @mcy looks like you have to re-approve the CI workflow for the latest push.

**Comment by mcy** [comment]
LG. Will merge once CI finishes.

**Pull request merged**